src/core/mac/sub_mac.cpp

```diff
@@ -1078,42 +1034,53 @@ const char *SubMac::CslStateToString(CslState aCslState)
 // CSL Receiver methods
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-void SubMac::SetCslChannel(uint8_t aChannel)
+void SubMac::CslSample(uint16_t aPeriod, uint8_t aChannel)
 {
-    mCslChannel = aChannel;
-}
+    Error error = kErrorNone;
 
-void SubMac::SetCslPeriod(uint16_t aPeriod)
-{
-    VerifyOrExit(mCslPeriod != aPeriod);
+#if OPENTHREAD_CONFIG_MAC_FILTER_ENABLE
+    VerifyOrExit(!mRadioFilterEnabled, error = Get<Radio>().Sleep());
+#endif
 
-    mCslPeriod = aPeriod;
+    SetState(kStateCslSample);
+
+    SetCslChannel(aChannel);
+
+    /* Only modify the timer when there is a CSL period change */
```

Comment:
I think `/* .. */` is mostly used in API doc. For implementation comments, probably let's use `//` to keep consistency.